File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        std::unordered_map<int, int> count_map;

        for (int i = 0; i < nums.size(); i++) {
            count_map[nums[i]]++;
        }


        std::vector<std::pair<int, int>> vec;
        for (auto & p : count_map) {
            vec.push_back(p);
        }

        std::sort(vec.begin(), vec.end(), [](auto &a, auto &b){
            return a.second > b.second;
        });

        vector<int> result;
        for (int i = 0; i < k; i++) {
            result.push_back(vec[i].first);
        }

        return result;
    }
};

// 1 2 2 3 3 3, k = 2 --> 3 is most freq, 2 is 2nd most freq
// 1 4 4 4 3 3 3 --> invalid input since not unique
//

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 0
        r = 0

        while r < len(nums):
            nums[l] = nums[r]
            while r < len(nums) and nums[r] == nums[l]:
                r += 1
            l += 1
        
        return l


# [1, 1, 2, 3, 4]
# [1, 2, 3, 4, x]              


# [2, 10, 10, 10, 30, 30, 30]
# [2, 10, 30,  x,  x,  x,  x]

# [2, 10, 10, 10, 30, 30, 30]
#                          r 

# [2, 10, 30]
#          l